src/resolvers/Mutation/removeOrganization.ts

```diff
@@ -54,124 +54,55 @@ export const removeOrganization: MutationResolvers["removeOrganization"] =
 
     // Remove organization._id from createdOrganizations list of currentUser.
     await User.updateOne(
       {
         _id: currentUser._id,
       },
       {
-        $set: {
-          createdOrganizations: currentUser.createdOrganizations.filter(
-            (createdOrganization) =>
-              createdOrganization.toString() !== organization._id.toString()
-          ),
+        $pull: {
+          createdOrganizations: organization._id,
         },
       }
     );
 
     // Remove organization._id from each member's joinedOrganizations field for organization.members list.
-    for (const memberId of organization.members) {
-      const member = await User.findOne({
-        _id: memberId,
-      }).lean();
-
-      await User.updateOne(
-        {
-          _id: memberId,
-        },
-        {
-          $set: {
-            joinedOrganizations: member?.joinedOrganizations.filter(
-              (joinedOrganization) =>
-                joinedOrganization.toString() !== organization._id.toString()
-            ),
-          },
-        }
-      );
-    }
+    await User.updateMany(
+      { _id: { $in: organization.members } },
+      { $pull: { joinedOrganizations: organization._id } }
+    );
 
     // Remove organization._id from each admin's joinedOrganizations field for organization.admins list.
-    for (const adminId of organization.admins) {
-      const admin = await User.findOne({
-        _id: adminId,
-      }).lean();
-
-      await User.updateOne(
-        {
-          _id: adminId,
-        },
-        {
-          $set: {
-            joinedOrganizations: admin?.joinedOrganizations.filter(
-              (joinedOrganization) =>
-                joinedOrganization.toString() !== organization._id.toString()
-            ),
-          },
-        }
-      );
-    }
+    await User.updateMany(
+      { _id: { $in: organization.admins } },
+      { $pull: { joinedOrganizations: organization._id } }
+    );
 
     /*
     Remove membershipRequest._id from each requester's membershipRequests
     field for membershipRequest.user for organization.membershipRequests list.
     */
-    for (const membershipRequestId of organization.membershipRequests) {
-      const membershipRequest = await MembershipRequest.findOneAndDelete({
-        _id: membershipRequestId,
-      }).lean();
-
-      const requester = await User.findOne({
-        _id: membershipRequest?.user,
-      }).lean();
-
-      await User.updateOne(
-        {
-          _id: requester?._id,
-        },
-        {
-          $set: {
-            membershipRequests: requester?.membershipRequests.filter(
-              (request) =>
-                request.toString() !== membershipRequest?._id.toString()
-            ),
-          },
-        }
-      );
-    }
+    await MembershipRequest.deleteMany({
+      _id: { $in: organization.membershipRequests },
+    });
+    await User.updateMany(
+      { _id: { $in: organization.membershipRequests } },
```

Comment:
This is wrong.

Reply:
Can you please elaborate a little bit sir. i was not able to get it!?

Reply:
@nishant3721 `organization.membershipRequests` is a list of `membershipRequestIds` not `userIds`.